pyatlan/events/atlan_event_handler.py

```diff
@@ -0,0 +1,151 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Atlan Pte. Ltd.
+from typing import Iterable, List, Optional
+
+from pyatlan.client.atlan import AtlanClient
+from pyatlan.model.assets import Asset, Catalog
+from pyatlan.model.events import AtlanEvent, AtlanEventPayload
+from pyatlan.model.search import DSL, Bool, IndexSearchRequest, Term
+
+WEBHOOK_VALIDATION_REQUEST = '{"atlan-webhook": "Hello, humans of data! It worked. Excited to see what you build!"}'
+
+
+def is_validation_request(data: str) -> bool:
+    return WEBHOOK_VALIDATION_REQUEST == data
+
+
+def valid_signature(expected: str, headers: dict[str, str]) -> bool:
+    """
+    Validate the signing secret provided with a request matches the expected signing secret.
+    """
+    if not headers:
+        return False
+    else:
+        found = headers.get("x-atlan-signing-secret")
+        return found is not None and found == expected
+
+
+def get_current_view_of_asset(
+    client: AtlanClient,
+    from_event: Asset,
+    limited_to_attributes: Optional[Iterable[str]] = None,
+    include_meanings: bool = False,
+    include_atlan_tags: bool = False,
+) -> Optional[Asset]:
+    """
+    Retrieve a limited set of information about the asset in Atlan,
+    as up-to-date as is available in the search index, to ensure we have
+    reasonably up-to-date information about it.
+    Note: this will be faster than getCurrentFullAsset, but relies on the eventual
+    consistency of the search index so may not have the absolute latest information about
+    an asset.
+    """
+    be_active = Term.with_state("ACTIVE")
+    be_of_type = Term.with_type_name(from_event.type_name)
+    have_qn = Term.with_qualified_name(from_event.qualified_name)
+    query = Bool(must=[be_active, be_of_type, have_qn])
+    dsl = DSL(query=query)
+    attributes = ["name", "anchor", "aws_arn"]
+    if limited_to_attributes:
+        attributes.extend(limited_to_attributes)
+    request = IndexSearchRequest(
+        dsl=dsl,
+        attributes=attributes,
+        relation_attributes=["guid"],
+        exclude_meanings=~include_meanings,
+        exclude_atlan_tags=~include_atlan_tags,
+    )
+    response = client.search(criteria=request)
+    if result := response.current_page()[0]:
+        return result
+    return None
+
+
+def has_description(asset: Asset) -> bool:
+    """
+    Check if the asset has either a user-provided or system-provided description.
+    """
+    description = asset.user_description
+    if not description:
+        description = asset.description
+    return description is not None and description != ""
```

Comment:
A bit more succinct:
`description = asset.user_description or asset.description`
`return description is not None and description != ""`